app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabase.kt

```diff
@@ -582,26 +636,163 @@ class FirebaseDatabase(reference: DatabaseReference = Firebase.database.referenc
+                if (snapshot.exists()) {
+                    // Get tournament data and check if corrupted
+                    val tournament = FirebaseDatabaseUtils.mapToTournament(snapshot)
+                    if (tournament == null) {
+                        throw Exception("Corrupted data for the tournament $tournamentId.")
+                    } else {
+                        tournamentValue.postValue(tournament!!)
+                    }
+                } else {
+                    throw Exception("The tournament with tournamentId $tournamentId doesn't exist.")
+                }
+            }
+        }
+        // add the listener on the database on the root of the tournament with tournament id
+        tournamentsKeysRoot().child(tournamentId).ref.addValueEventListener(tournamentValueEventListener)
+        return tournamentValue
+    }
+
+    override fun getTournamentPosts(tournamentId: String): LiveData<List<TournamentPost>> {
+        val tournamentPostsValue = MutableLiveData<List<TournamentPost>>()
+        val tournamentPostsValueEventListener: ValueEventListener
+
+        // listener for data change
+        tournamentPostsValueEventListener = object : ValueEventListener {
+            override fun onCancelled(error: DatabaseError) {
+                throw Exception("The tournament with tournamentId $tournamentId can't be fetched from the database.")
+            }
+
+            override fun onDataChange(snapshot: DataSnapshot) {
+                // Check that the tournament exists
+                if (snapshot.exists()) {
+                    // Get tournament posts
+                    val posts = FirebaseDatabaseUtils.transformPostList(snapshot)
+                    tournamentPostsValue.postValue(posts)
+                } else {
+                    throw Exception("The tournament with tournamentId $tournamentId doesn't exist.")
+                }
+            }
+        }
+        // add the listener on the database on the root of the tournament posts with tournament id
+        tournamentsRoot().child(tournamentId).child(FirebaseKeys.TOURNAMENT_POSTS).ref.addValueEventListener(tournamentPostsValueEventListener)
+        return tournamentPostsValue
+    }
+
+    override fun getTournamentParticipantsId(tournamentId: String): CompletableFuture<List<String>> {
+        val future = CompletableFuture<List<String>>()
         // Check that the tournament exists
         isTournamentInDatabase(tournamentId).thenApply { tournamentExists ->
             if (!tournamentExists) {
                 future.completeExceptionally(Exception("The tournament with tournamentId $tournamentId doesn't exist."))
             } else {
-                // Get tournament data and check if corrupted
-                tournamentsRoot().child(tournamentId).get().addOnSuccessListener {
-                    val tournament = FirebaseDatabaseUtils.mapToTournament(it)
+                // Get tournament posts
```

Comment:
I think there is a type , here we get the tournaments participants and not the posts.

Reply:
Absolutely!